### ISSUE TITLE
draw.delete() trigger no event but draw.render

### ISSUE THREAD

**Issue by Youllou** [mapbox/mapbox-gl-draw#1105]
Hi,

**mapbox-gl-js version**: v2.8.1
**mapbox-gl-draw version**:v1.3.0

### Steps to Trigger Behavior

 1. make a listener for every draw event that log it's event name
 2. draw an element
 3. delete it from a command (`draw.delete(id)` or `draw.deleteAll()`)
 4. open the console
(I recommend adding a little log that says that you deleted so you know when what was fired)

### Expected Behavior

If I understood the documentation well, the delete function should, at least, trigger the [`draw.selectionchange`](https://github.com/mapbox/mapbox-gl-draw/blob/main/docs/API.md#drawselectionchange)
![image](https://user-images.githubusercontent.com/62644404/165317844-a152be49-0dd3-441e-873f-51700e432738.png)
[...]
![image](https://user-images.githubusercontent.com/62644404/165317923-7923cc72-c9b4-44bb-9615-43fccf5361f1.png)

But I think that `draw.delete` and maybe even `draw.update` events should be triggered

### Actual Behavior

As I logged every draw event, I've seen that the only triggered event was `draw.render`

### Example
I have made a gist that recreates this issue
https://gist.github.com/Youllou/838015098d9cbdcc8f2aa859ee75c337

**Comment by tordans** [comment]
@SnailBones did you find a work around for this?

**Comment by Youllou** [comment]
@tordans If I remember well, I catched the event from the draw.render event then check wether or not there was a selection.

**Comment by Youllou** [comment]
**TLDR** : this is not a bug, if you want the api to trigger events you should add the `suppressAPIEvents` boolean to the options when creating the `MapboxDraw` object

---
Alright so quick update on this

After a quick reading of the api, I thought I understood why it was like that and saw a quick fix. The Mapbox gl Draw api uses a silent boolean to tell to the context store whether the api should fire events or not
![image](https://github.com/user-attachments/assets/a113970b-e158-4f28-bc1b-b88c6680647f)

It is then used each time the context store is modified 
for add :
![image](https://github.com/user-attachments/assets/3ba855a0-f019-41d5-b898-abea52c6fccb)
for delete :
![image](https://github.com/user-attachments/assets/289e427b-799c-4f0a-8375-f65bf8941dc7)

This is interesting but not well documented at all.
So to change this you simply need to add a `suppressAPIEvents: false` in the options when creating the MabpoxDraw object just like that :
```js
const modes = MapboxDraw.modes;
const Draw = window.Draw = new MapboxDraw({ modes, suppressAPIEvents: false });
```

I will keep this issue open for now but might close it in favor of an issue that asks for better documentation on that point

**Comment by Youllou** [comment]
This has, in fact, been changed in #1264 since december 2024 
From my understanding this was an issue prior to this PR but it has been fixed.
Not sure, anyways it is fixed/never was an issue ! Yipee !
